### ISSUE TITLE
using put method with buffer doesn't work

### ISSUE THREAD

**Issue by ggprod** [theophilusx/ssh2-sftp-client#561]
I attempted to use the put method with a buffer as described here: https://www.npmjs.com/package/ssh2-sftp-client#orgc6243de

but it does not work and throws an error: `Re-thrown: Cannot read properties of undefined (reading 'once')`

**Comment by theophilusx** [comment]
Most likely cause is an error in your code. 

Based on the information provided, I cannot add anything else.

**Comment by ggprod** [comment]
yes, apologies, I traced this problem to another library that had erroneous output
